packages/now-cli/src/util/dev/server.ts

```diff
@@ -630,13 +626,20 @@ export default class DevServer {
     this.apiExtensions = detectApiExtensions(config.builds || []);
 
     // Update the env vars configuration
-    const [runEnv, buildEnv] = await Promise.all([
+    let [runEnv, buildEnv] = await Promise.all([
       this.getLocalEnv('.env', config.env),
       this.getLocalEnv('.env.build', config.build?.env),
     ]);
-    const allEnv = { ...buildEnv, ...runEnv };
-    this.envConfigs = { buildEnv, runEnv, allEnv };
 
+    let allEnv = { ...buildEnv, ...runEnv };
+
+    // If no .env/.build.env is present, fetch and use cloud environment variables
+    if (Object.keys(allEnv).length === 0) {
```

Comment:
I think this will never be true because `NOW_REGION` is always added, right?

Reply:
I would suggest adding `NOW_REGION` to `runEnv` after reading the files and reading the cloudEnv.

Something like this:

```js
if (Object.keys(allEnv).length === 0) {
  const cloudEnv = this.populateVercelEnvVars(this.environmentVars);
  runEnv = buildEnv = cloudEnv;
}
runEnv['NOW_REGION'] = 'dev1';
const allEnv = { ...buildEnv, ...runEnv };
this.envConfigs = { buildEnv, runEnv, allEnv };
```

Reply:
Nevermind, ignore my comments